clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java

```diff
@@ -135,22 +144,28 @@ public void append(K key, ByteBuffer element) {
 
   }
 
-  public boolean replaceAtHead(K key, Chain expected, Chain replacement) {
+  public void replaceAtHead(K key, Chain expected, Chain replacement) {
     final Lock lock = heads.writeLock();
     lock.lock();
     try {
-      InternalChain chain = heads.get(key);
-      if (chain == null) {
-        if (expected.iterator().hasNext()) {
-          return false;
+      while (true) {
+        InternalChain chain = heads.get(key);
+        if (chain == null) {
+          if (expected.iterator().hasNext()) {
```

Comment:
`!expected.isEmpty()` ?  Seems a waste to create an `Iterator` that's not going to be used.